File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string result = strs[0];
        int n = strs.size();
        int len = result.length();
     
        for (int i = 1; i < n; i++) {
            while (strs[i].find(result) != 0) {
                result = result.substr(0, len - 1);
                len--;
                if (result.empty()) {
                    return result;
                }
            }
        }

        return result;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    const len = strs.length;

    if (!len) {
        return "";
    }

    if (len == 1) {
        return strs[0];
    }

    let pref = strs[0];

    for (var i = 1; i < len; i++) {
        while (!strs[i].startsWith(pref)) {
            pref = pref.slice(0, -1);
        }

        if (pref == "") {
            return "";
        }
    }

    return pref;
};